File: ride-service/src/main/java/com/example/ride_service/model/event/RideCancelledEvent.java
package com.example.ride_service.model.event;

import com.example.ride_service.model.enums.CancelInitiator;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record RideCancelledEvent(
        UUID rideId,
        UUID driverId,
        CancelInitiator initiator,
        LocalDateTime cancelAt
) {
}

File: driver-service/src/main/kotlin/com/example/driver_service/model/event/RideCancelledEvent.kt
package com.example.driver_service.model.event

import com.example.driver_service.model.enums.CancelInitiator
import java.time.LocalDateTime
import java.util.UUID

data class RideCancelledEvent(
    val rideId: UUID,
    val driverId: UUID,
    val initiator: CancelInitiator,
    val cancelAt: LocalDateTime
) {
}
